lib/setup-terraform.js

```diff
@@ -183,40 +113,35 @@ async function run () {
     const wrapper = core.getInput('terraform_wrapper') === 'true';
 
     // Gather OS details
-    const osPlat = os.platform();
+    const osPlatform = os.platform();
     const osArch = os.arch();
 
-    // Download metadata about all versions of Terraform CLI
-    const versionMetadata = await downloadMetadata();
-
-    const specificMatch = findSpecific(versionMetadata, version);
-    // Find latest or a specific version like 0.1.0
-    const versionObj = version.toLowerCase() === 'latest'
-      ? findLatest(versionMetadata) : specificMatch || findLatestMatchingSpecification(versionMetadata, version);
-
-    if (versionObj) {
-      // Get the build available for this runner's OS and a 64 bit architecture
-      const buildObj = getBuild(versionObj, osPlat, osArch);
+    core.debug(`Finding releases for Terraform version ${version}`);
+    const release = await releases.getRelease('terraform', version);
+    const platform = mapOS(osPlatform);
+    const arch = mapArch(osArch);
+    core.debug(`Getting build for Terraform version ${release.version}: ${platform} ${arch}`);
+    const build = release.getBuild(platform, arch);
+    if (!build) {
+      throw new Error(`Terraform version ${version} not available for ${platform} and ${arch}`);
+    }
 
-      // Download requested version
-      const pathToCLI = await downloadCLI(buildObj.url);
+    // Download requested version
+    const pathToCLI = await downloadCLI(build.url);
 
-      // Install our wrapper
-      if (wrapper) {
-        await installWrapper(pathToCLI);
-      }
+    // Install our wrapper
+    if (wrapper) {
+      await installWrapper(pathToCLI);
+    }
 
-      // Add to path
-      core.addPath(pathToCLI);
+    // Add to path
+    core.addPath(pathToCLI);
 
-      // Add credentials to file if they are provided
-      if (credentialsHostname && credentialsToken) {
-        await addCredentials(credentialsHostname, credentialsToken, osPlat);
-      }
-      return versionObj;
-    } else {
-      core.setFailed(`Could not find Terraform version ${version} in version list`);
+    // Add credentials to file if they are provided
+    if (credentialsHostname && credentialsToken) {
+      await addCredentials(credentialsHostname, credentialsToken, osPlatform);
     }
+    return release;
```

Comment:
I may have lost this change in the context of the PR, just want to confirm we don't want to return `versionObj`